### PR TITLE
Stop splitting common operators

### PR DESCRIPTION
**Pull request by Jentsch**
This PR fixes the operators like `->`, `:+`, `=>`, `<-` that are 'composed' from shorter operators like `-` and `>`. It also fixes operators like `\/` where `\` wasn't highlighted at all.

The change looks quite drastic, but the changes in the snaps look fine to me. I'd would appraciate a second pair of eye to check if an special operator is missing.

A change where I'm not sure if we want it:
```
   i -= 1
//  was:
//   ^ keyword.operator.arithmetic.scala
//    ^ keyword.operator.comparison.scala
//  now:
//   ^^ keyword.operator.scala <- 
```
One could argue, the `-` is realy the arithmetic operator here, as the
expressions `i -= 1` most of the time means `i = i - 1`.

Btw. I don't think in `i = 1` the `=` shouldn't be marked as a comparison but just as an operator.

### CODE REVIEW

**Comment by bishabosha** [comment]
I think that because you are allowed to define `def -= (foo: Int) ...` that the interpretation is fine

**Comment by Jentsch** [comment]
I changed how operators are recognized and I'll update the PR text.

File: src/typescript/Scala.tmLanguage.ts [hunk 1]
```diff
@@ -677,21 +676,55 @@ export const scalaTmLanguage: TmLanguage = {
...
+                {
+                  match: '(\\!)',
+                  name: 'keyword.operator.logical.scala'
+                },
+                {
+                  match: '(\\*|-|\\+|/|%|~)',
+                  name: 'keyword.operator.arithmetic.scala'
+                },
+                {
+                  match: '(=|\\<|>)',
```
Comment (Jentsch) [inline src/typescript/Scala.tmLanguage.ts:718]: Here I think we shouln't match `=`, as a single `=` doesn't compare but defines things.

**Review by tgodzik — CHANGES_REQUESTED** [review]
Makes sense! You need to rebase on main to get this mergable though

**Comment by Jentsch** [comment]
Done

**Review by tgodzik — APPROVED** [review]
Thanks!

**Pull request merged**